gatsby-config.js

```diff
@@ -15,103 +15,120 @@ const tracking = {
   heap: "3368297951",
 }
 
-module.exports = {
-  siteMetadata: {
-    title: 'Gatsby Contentful TypeScript starter',
-    enableDDO: process.env.ENABLE_DDO === '1'
-  },
-  plugins: [
-    'gatsby-plugin-react-helmet',
-    {
-      resolve: `gatsby-plugin-manifest`,
+const conditionallyAddAlgoliaPlugin = (plugins) => {
+  if (process.env.ENABLE_ALGOLIA_INDEXING === '1') {
+    const queries = require("./src/util/algolia.js")
+    plugins.push({
+      resolve: `gatsby-plugin-algolia`,
       options: {
-        name: 'gatsby-starter-contentful-typescript',
-        short_name: 'starter',
-        start_url: '/',
-        background_color: '#663399',
-        theme_color: '#663399',
-        display: 'minimal-ui',
-        icon: 'src/img/brand/favicon-96x96.png', // This path is relative to the root of the site.
-      },
-    },
-    {
-      resolve: 'gatsby-plugin-i18n',
-      options: {        
-        langKeyDefault: 'en',
-        useLangKeyLayout: false
+        appId: process.env.ALGOLIA_APP_ID,
+        apiKey: process.env.ALGOLIA_ADMIN_KEY,
```

Comment:
Hmm, so it seems like if I just copied `.env.sample` to `.env`, the default behavior would enable Algolia indexing, but pass empty strings for `appId` and `apiKey` to the Algolia plugin, which I'm guessing wouldn't make the plugin very happy, probably causing it to raise a weird hard-to-debug error?  Should we further conditionally call this code if indexing is enabled _and_ `ALGOLIA_APP_ID` and `ALGOLIA_ADMIN_KEY` are non-empty?  Or alternatively, we could _disable_ Algolia indexing by default and raise a friendly error if it's enabled but the app id and/or admin key are undefined... up to you really, I just want to make sure devs who are onboarding can have a build that works with minimal configuration!

Reply:
Hmm, another option is to simply disable Algolia indexing if the algolia app id and/or admin key aren't defined?  That would reduce the configuration complexity, and that way netlify test branches/deploys could simply not define those credentials/set them to empty values, which could also cut down on potential security concerns from untrusted branches (a danger of putting sensitive credentials in a CI/build environment is that if a volunteer submits a malicious PR, they could steal the credentials and do nefarious things with them)... not that it's particularly likely or anything, heh.

Reply:
Got it! That sounds like a good idea. I updated accordingly ^^

Reply:
For now, I'm still going to keep the Algolia App Id and Search Key as required variables, as these will be needed to actually conduct searches on our Algolia index in the search bar itself. Once we have the UI built out, we can maybe try and make the entire search bar component optional, in which case these variables can also be optional and help determine whether to render the search bar or not.